portalnet/src/overlay_service.rs

```diff
@@ -487,6 +493,15 @@ where
                                 );
                             }
                         }
+                        OverlayCommand::RequestEventStream(callback) => {
```

Comment:
Am I missing something? This is only called once, iiuc. Why do we need to include it inside the `select` loop? Could we not just instantiate it once at the beginning of the `start()` method.

Reply:
This is a command that the overlay service receives from `OverlayProtocol` to create an event stream.

 If we include the creation of the event stream at the beginning of the `start()` method, it will create this stream for any `OverlayProtocol` instance, including tests, without giving us the opportunity to choose for which instance we want to create the stream.  The API is as follows:

 ```rust
 let overlay = OverlayProtocol::new(...);
 // Here we explicitly create the stream and we can pass it to another service if needed.
let stream = overlay.event_stream().await?'
````

For example, in this PR, we create the stream only for the Beacon Network because we will need the overlay protocol to signalize to the beacon service as soon as the node receives new `LightClientOptimisticUpdate` or `LightClientFinalityUpdate` content items.